credential-exchange-types/src/format.rs

```diff
@@ -217,6 +197,115 @@ pub struct PasskeyCredential {
     pub fido2_extensions: Option<Fido2Extensions>,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreditCardCredential {
+    pub number: String,
+    pub full_name: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub card_type: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub verification_number: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub expiry_date: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub valid_from: Option<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NoteCredential {
+    pub content: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TotpCredential {
+    pub secret: B32,
+    pub period: u8,
+    pub digits: u8,
+    pub username: String,
+    pub algorithm: OTPHashAlgorithm,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub issuer: Option<String>,
+}
```

Comment:
**Suggestion:** These should probably contain documentation as well

Reply:
They should, but they were still pending PRs in the spec.